### PR TITLE
AMA-700 - syllabus rejigger python migration

### PR DESCRIPTION
**Pull request by NgoniChoga**

### CODE REVIEW

File: work/syllabus_rejigger.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/python3
+
+## This script takes as input the syllabus-archive.xml file inside the site-archive folder.
```
Comment (smarquard) [inline work/syllabus_rejigger.py:3]: syllabus.xml

File: work/syllabus_rejigger.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+            asset_node.set("syllabus_body-html", base64.b64encode("".join(content).encode("utf-8")).decode("utf-8"))
+
+    # Write the modified XML to a new file
+    logging.info(f'Site {SITE_ID} syllabus data has been updated.')
+    tree.write(xml_src,  encoding='utf-8', xml_declaration=True)
+
+
+def main():
+    APP = config.config.APP
+    parser = argparse.ArgumentParser(description="This script applies an HTML template to syllabus-archive.xml",
```
Comment (smarquard) [inline work/syllabus_rejigger.py:82]: syllabus.xml

File: requirements.txt [hunk 3]
```diff
@@ -6,8 +6,8 @@ Jinja2==3.1.3
 jira==3.4.1
 jproperties==2.1.1
 jsonpath_ng==1.6.0
-lxml==4.9.1
-numpy==1.22.4
+lxml==4.9.4
+numpy==1.26.4
```
Comment (smarquard) [inline requirements.txt:10]: We don't actually need numpy, so dropped it in https://github.com/cilt-uct/Brightspace-migrate-from-sakai/commit/33a1e813da1fbdbdaf4aade23c62873aa2fc5c58

File: tests/test_syllabus_rejigger.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+    def check_converted(self, xml_src: str):
+        tree = ET.parse(xml_src)
+        root = tree.getroot()
+
+        # Find syllabus_data nodes
+        syllabus_data = root.findall(".//siteArchive/syllabus/syllabus_data/asset")
+        encoded_html = syllabus_data[0].get("syllabus_body-html")
+
+        original = "PHAgc3R5bGU9InRleHQtYWxpZ246IGNlbnRlcjsiPjxpbWcgYWx0PSIiIGhlaWdodD0iMTExNiIgc3JjPSJodHRwczovL3Z1bGEudWN0LmFjLnphL2FjY2Vzcy9jb250ZW50L2dyb3VwLzlhYWVkYTI2LTU5YjgtNGVjMi04YzE5LTk1Nzk3ZmE4YWE0My9BZG1pbi9MZXNzb24lMjBwbGFuJTIwdjMucGRmIiB3aWR0aD0iMjAwMCIgLz48L3A+CgoKPGgzPkF0dGFjaG1lbnRzPC9oMz4KCjx1bD4KPGxpPjxhIGhyZWY9Imh0dHBzOi8vdnVsYS51Y3QuYWMuemEvYWNjZXNzL2NvbnRlbnQvYXR0YWNobWVudC85YWFlZGEyNi01OWI4LTRlYzItOGMxOS05NTc5N2ZhOGFhNDMvTGVzc29uIFBsYW4vNzZlMDk2YTQtMjBkMC00YzVmLWI1ZTYtNWY0MDYzOTYyMjg1L0xlc3NvbiBwbGFuIHYzLnBkZiI+TGVzc29uIHBsYW4gdjMucGRmPC9hPjwvbGk+CjwvdWw+Cg=="
```
Comment (smarquard) [inline tests/test_syllabus_rejigger.py:38]: This test would be a bit clearer about what it's supposed to do if the original / new comparison was the non-encoded version (readable html)

Comment (NgoniChoga) [reply tests/test_syllabus_rejigger.py:45]: updated

**Pull request merged**